rust/src/mask/masking.rs

```diff
@@ -101,6 +131,20 @@ impl Aggregation {
         Ok(())
     }
 
+    /// Unmasks the aggregated masked model with the given `mask`.
+    ///
+    /// It should be checked that [`validate_unmasking()`] returns `true` before calling this, since
```

Comment:
Not related to this pr:
Was there a reason to not include the validation in the unmask method?

Reply:
it was combined in the first implementation, but was separated in the refactoring. the reason is twofold, first this provides some logical and visual separation (one can of course argue if this is needed at all since the methods are not that complex). and second this allows for atomic changes where all the conditions have to be checked first before any action is taken (this is not the case for unmasking directly, but is for example needed for aggregation, and this way we have consistency in the method style).

Reply:
Ok that makes sense. Thanks for the explanation!